File: python/exercise46/word_frequency_finder.py
from flask import Flask

app = Flask(__name__)

store: dict[str, int] = dict()


def count_words(words: list[str]):
    for word in words:
        store[word] = words.count(word)


with open("./words.txt") as file:
    contents = file.read().replace("\n", " ").strip()

    stripped = contents.translate(
        {ord(i): None for i in ",;:_[]?!@\"'.—”“’)(&*%$#-|I\n\t"}
    )
    # stripped = contents.strip(",;:_[]?!@\"'.—”“’)(&*%$#-|I\r\n")
    splitted = stripped.split(" ")


@app.get("/")
def index():
    count_words(splitted)
    return "<h1>Hello again Flask!!</h1>"

File: javascript/exercise46/word-frequency-finder.js
import fs from "node:fs";
import { Hono } from "hono";
import { serve } from "@hono/node-server";

const app = new Hono();

/** @type {string} contents */
const contents = fs.readFileSync("./macbeth.txt", "utf8").trim();

/** @type {Map<string, number>} store */
const store = new Map();

const splitted = contents.split(/ |\n/);

for (const word of splitted) {
	const trimmed = word.replace(/[,;:_?!@\"'.—”“’\)\(&*%$#-|I],/g, "");

	if (trimmed === "") {
		continue;
	}

	store.set(trimmed, store.get(trimmed) + 1 || 1);
}

console.log(Object.fromEntries(store));

app.get("/", (c) => {
	return c.render("<h1>Hello word finder!!</h1>");
});

serve({ fetch: app.fetch, port: 5000 });
